### PR TITLE
Trend micro vision one

### PR DESCRIPTION
**Pull request by shaqnawe**
## Contributing to Cortex XSOAR Content
Make sure to register your contribution by filling the [contribution registration form](https://forms.gle/XDfxU4E61ZwEESSMA)

**The Pull Request will be reviewed only after the contribution registration form is filled.**

## Status
- [x] In Progress
- [ ] Ready
- [ ] In Hold - (Reason for hold)

## Related Issues
fixes: link to the issue

## Description
A few sentences describing the overall goals of the pull request's commits.

## Must have
- [x] Tests
- [x] Documentation

### CODE REVIEW

File: Packs/TrendMicroVisionOne/Integrations/TrendMicroVisionOneV3/TrendMicroVisionOneV3.py [hunk 1]
```diff
@@ -2703,6 +2712,78 @@ def delete_custom_script(
...
+        ingested_start_date_time=ingested_start_time,
+        ingested_end_date_time=ingested_end_time,
+        top=top,
+        op=query_op,
+        **fields,
+    )
+    # Check if an error occurred during rest call
+    if _is_pytmv1_error(resp.result_code):
+        err: pytmv1.Error = unwrap(resp.error)
+        return_error(message=f"{err.message}", error=str(err))
```
Comment (Shellyber) [inline Packs/TrendMicroVisionOne/Integrations/TrendMicroVisionOneV3/TrendMicroVisionOneV3.py:2754]:
```suggestion
        raise Exception(message=f"{err.message}", error=str(err))
```

File: Packs/TrendMicroVisionOne/Integrations/TrendMicroVisionOneV3/TrendMicroVisionOneV3.py [hunk 2]
```diff
@@ -2703,6 +2712,78 @@ def delete_custom_script(
...
+    resp_type: pytmv1.ListOatsResp = unwrap(resp.response)
+    if resp_type.total_count > 50:
+        return_error("Please refine search, this query returns more than 50 results.")
```
Comment (Shellyber) [inline Packs/TrendMicroVisionOne/Integrations/TrendMicroVisionOneV3/TrendMicroVisionOneV3.py:2757]:
```suggestion
        raies Exception("Please refine search, this query returns more than 50 results.")
```

File: Packs/TrendMicroVisionOne/pack_metadata.json [hunk 3]
```diff
@@ -2,7 +2,7 @@
     "name": "Trend Micro Vision One",
     "description": "Trend Micro Vision One is a purpose-built threat defense platform that provides added value and new benefits beyond XDR solutions, allowing you to see more and respond faster. Providing deep and broad extended detection and response(XDR) capabilities that collect and automatically correlate data across multiple security layers—email, endpoints, servers, cloud workloads, and networks—Trend Micro Vision One prevents the majority of attacks with automated protection.",
     "support": "partner",
-    "currentVersion": "4.1.1",
+    "currentVersion": "4.2.0",
```
Comment (Shellyber) [inline Packs/TrendMicroVisionOne/pack_metadata.json:5]:
```suggestion
    "currentVersion": "4.1.2",
```
Please update the release notes file name accordingly

**Review by Shellyber — CHANGES_REQUESTED** [review]
Looks good!
Thank you for your contribution
Please see my notes

File: Packs/TrendMicroVisionOne/Integrations/TrendMicroVisionOneV3/TrendMicroVisionOneV3.yml [hunk 4]
```diff
@@ -1301,7 +1301,58 @@ script:
...
+      type: string
+    - contextPath: VisionOne.Get_Observed_Attack_Techniques.detected_date_time
+      description: Timestamp in ISO 8601 format that indicates when an Observed Attack Techniques event was detected.
+      type: string
+    - contextPath: VisionOne.Get_Observed_Attack_Techniques.ingested_date_time
+      description: Timestamp in ISO 8601 format that indicates when the pipeline ingested data related to an Observed Attack Techniques event. This field is displayed only when ingestedStartDateTime and ingestedEndDateTime are used to define the data retrieval time range.
+      type: string
+    description: Displays a list of Observed Attack Techniques events that match the specified criteria.
+    name: trendmicro-visionone-get-observed-attack-techniques
+  dockerimage: devtesting/pytmv1:0.8.3.1715976040
```
[the remark this answers is not in the document]
Comment (kgal-pan) [reply Packs/TrendMicroVisionOne/Integrations/TrendMicroVisionOneV3/TrendMicroVisionOneV3.yml:1355]:
```suggestion
  dockerimage: demisto/pytmv1:0.8.3.96268
```

**Comment by Shellyber** [comment]
Hi @shaqnawe ,
Thank you for your contribution.
I have not received your response in the last two weeks, so I will close your PR now.
Please feel free to re-open it whenever you're available.
Thanks again.

**Pull request closed without merging**